middleware/RetryHandler.md

```diff
@@ -10,8 +10,9 @@ Provide a reusuable component that provides application developers with effectiv
 - A `retry-attempt` header SHOULD be added which contains a numeric value representing the retry number.  This value will be used for diagnostics and determining the effectiveness of the retry handler.
 - Retries MUST respect the `retry-after` response header if provided.
 - Where no `retry-after` header is provided by the server, an exponential backoff with random offset hueristic should be used to determine the retry delay.
-- Retries should be limited to a maximum value.
-- Only requests with payloads that are buffered/rewindable are supported.  Payloads with forward only streams will be have the responses returned without any retry attempt.
+- Customers can specify a custom value for the `RetriesTimeLimit` greater than 0 to introduce time-based evaluated request retries alongside the default count-based request retry. The default is set at 0 second.
+- If the `RetriesTimeLimit` value has been specified, the cumulative retry time and `retry-after` value for each request retry will be evaluated against this value; if either the cumulative retry time or `retry-after` value is greater than the `RetriesTimeLimit`, the failed response will be immediately returned, else the request retry continues. This is applicable to both 429 and 503/504.
```

Comment:
```suggestion
- If the `RetriesTimeLimit` value has been specified, the cumulative retry time and `retry-after` value for each request retry will be evaluated against this value; if the cumulative retry time plus the `retry-after` value is greater than the `RetriesTimeLimit`, the failed response will be immediately returned, else the request retry continues. This is applicable to both 429 and 503/504.
```

Reply:
@darrel why are we evaluating the `RetriesTimeLimit` against the summation of the cumulative retry time and the `retry-after` value?

Reply:
Consider the scenario where we have received our 2nd 429.  The retry-after value 30 secs.  We have already waited 40 seconds.  5 secs for the first call to get the 429, 30 secs delay and then another 5 seconds to get the 429 for the second time.   Our RetriesTimeLimit is 60 seconds.  At 40 seconds we need to decide if we are going to wait another 30 seconds before trying again.  Based on the original wording that says "either the cumulative retry time or the retry-after value is greater than the RetriesTimeLimit" we would retry because 60 > 40 and 60 > 30.  However, we should not retry because 40secs + 30 secs is 70 secs, which means we would only be retrying for the second time at 70 secs which is after our RetriesTimeLimit.

Here's that scenario presented in a more coherent way...
RetriesTimeLimit = 60 sec

| Call | Duration (secs) | Response | Retry After (secs) | Cumulative Retry Time (secs) | Retry (Y/N) |
|--|--|--|--|--|--|
|GET /foo | 5  | 429 | 30 | 5 | Y  (30 + 5 < 60) |
|GET /foo | 5  | 429 | 30 | 40 | N (30 + 40 > 60 |